### PR TITLE
Add test

### PR DESCRIPTION
**Pull request by ocefpaf**
@robintw this crashed as you reported in #5 in my local tests. There is probably something wrong with this package and the compiler used in `conda-forge`, not sure how to fix that though :-/

I can remove the latest package create but, as I mentioned in #5, this will only postpone the problem and the past versions of the package cannot be used with used with the modern stack of packages in `conda-forge`.

### CODE REVIEW

**Comment by robintw** [comment]
Thank you for adding the test. Interestingly it seems be failing on Windows too (I didn't have a Windows machine handy to test on)

I managed to install the previous version of this package in my test environment - to check it worked - but I didn't have many other packages installed in that environment, so I didn't come across any of the conflicts this would cause with other conda-forge packages.

I'm not sure what the best way forward is then: obviously there's something strange going on with the compilers and this code. I've compiled the 6S code manually on many systems, and never come across this problem before.

Do you know of any experts in the conda-forge compiling system who could advise? Anyone in conda-forge who is an expert in Fortran compilation under the conda compilers? Or is it best to remove this version of the package for the moment, while we try and sort it out?

**Comment by ocefpaf** [comment]
> Do you know of any experts in the conda-forge compiling system who could advise?

@isuruf :grimacing: 

> Anyone in conda-forge who is an expert in Fortran compilation under the conda compilers?

Fortran? Sorry @isuruf, 2 in one :smile: 

> Or is it best to remove this version of the package for the moment, while we try and sort it out?

It is probably better to remove the ones that are creating the segfault. I'll do that in a moment.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,18 +1,15 @@
-{% set name = "sixs" %}
-{% set version = "1.1" %}
-{% set sha256 = "4253664bfed35a659ac602d4472780df7c665d67021d495327f47eb7325a2d7e" %}
+{% set version = "1.1.2" %}
```
Comment (ocefpaf) [inline recipe/meta.yaml:1]: A quick test locally with the latest version shows the same segault :-(

**Comment by isuruf** [comment]
It's got something to do with the default `FFLAGS` set. If I unset them, the build works fine.
It would be helpful if someone could figure out exactly which flag is problematic. (by doing a bisection search?)

**Comment by ocefpaf** [comment]
> It's got something to do with the default `FFLAGS` set. If I unset them, the build works fine.
> It would be helpful if someone could figure out exactly which flag is problematic. (by doing a bisection search?)

Thanks @isuruf. I'll give it a try tomorrow.

**Comment by robintw** [comment]
Thanks both - that's really helpful.

Just a quick note @isuruf: when this is all fixed, we will need to reinstate the `-ffpe-summary=none` flag in `FFLAGS` (I saw you'd commented it out), as without this the Py6S interface to 6S (which is how 6S is often used) won't work properly.

As a slight aside: can anyone enlighten me about what the various automated builds are doing? All of the `linux linux_`, `win win_` and `osx osx_` builds have failed, but looking at the build logs, they all look like they're running on some sort of Linux, even the `win win_` one. The AppVeyor build, which definitely looks like Windows, seems to have passed, suggesting that it does work on Windows.

Similarly, the latest build works fine on my OS X machine, but the `osx osx_` build still seems to have failed.

**Comment by ocefpaf** [comment]
> Thanks both - that's really helpful.

Looks like the segfault was caused by the `fopenmp`.

> Just a quick note @isuruf: when this is all fixed, we will need to reinstate the `-ffpe-summary=none` flag in `FFLAGS` (I saw you'd commented it out), as without this the Py6S interface to 6S (which is how 6S is often used) won't work properly.

Note that the latest commit has that flag but macOS is still failing. The error is:

```
x86_64-apple-darwin13.4.0-gfortran: error: unrecognized command line option ‘-ffpe-summary=none’
make[2]: *** [CMakeFiles/sixs.dir/src/main.f.o] Error 1
make[2]: *** Waiting for unfinished jobs....
x86_64-apple-darwin13.4.0-gfortran: error: unrecognized command line option ‘-ffpe-summary=none’
make[2]: *** [CMakeFiles/sixs.dir/src/AATSR.f.o] Error 1
make[1]: *** [CMakeFiles/sixs.dir/all] Error 2
make: *** [all] Error 2
```
 
> As a slight aside: can anyone enlighten me about what the various automated builds are doing? All of the `linux linux_`, `win win_` and `osx osx_` builds have failed, but looking at the build logs, they all look like they're running on some sort of Linux, even the `win win_` one. The AppVeyor build, which definitely looks like Windows, seems to have passed, suggesting that it does work on Windows.

You can ignore Azure for now but those are not all Linuxes, it is Windows and OS X as expected.

> Similarly, the latest build works fine on my OS X machine, but the `osx osx_` build still seems to have failed.

It is probably a difference in our compilers.

**Comment by robintw** [comment]
Ok, I think we can just remove the `-ffpe-summary=none` flag for OS X, as it seems the compiler there doesn't support it. Is there a way to remove a flag just for OS X?

Are we expecting the other Azure Pipelines builds to pass before we can merge this?

**Comment by ocefpaf** [comment]
> Ok, I think we can just remove the `-ffpe-summary=none` flag for OS X, as it seems the compiler there doesn't support it. Is there a way to remove a flag just for OS X?

With an if clause. See my last commit.

> Are we expecting the other Azure Pipelines builds to pass before we can merge this?

We are using Azure now everywhere but Windows. So we can merge this if Azure-Windows fails but we need to fix anything else.

**Comment by ocefpaf** [comment]
I had to patch `CMakeLists.txt` to remove `ffpe-summary`. @robintw this is ready for a review and merge if you are OK with this solution.

**Comment by robintw** [comment]
Yes, this looks fine to me.

Thank you everyone for your help with this.

**Comment by robintw** [comment]
I've just realised that I never actually merged this PR - oops!

Now there has been a rebuild for the gfortran update, it has created a new build that now fails on Linux.

Unfortunately I can't just merge this as there are lots of conflicts due to changes to the base feedstock (I think). Is it possible for you to do a rebase or something @ocefpaf, so I can merge this and hopefully build a version that works?

**Comment by ocefpaf** [comment]
> I've just realised that I never actually merged this PR - oops!
> 
> Now there has been a rebuild for the gfortran update, it has created a new build that now fails on Linux.
> 
> Unfortunately I can't just merge this as there are lots of conflicts due to changes to the base feedstock (I think). Is it possible for you to do a rebase or something @ocefpaf, so I can merge this and hopefully build a version that works?

I rebased this and it should be ready to go. It would be nice to get a license file for it. I could not find one upstream.

**Comment by robintw** [comment]
Thanks @ocefpaf - that's great.

The Windows build seems to be giving the following error though:

```
  sh.exe was found in your PATH, here:

  C:/Program Files/Git/bin/sh.exe

  For MinGW make to work correctly sh.exe must NOT be in your path.

  Run cmake from a shell that does not have sh.exe in your PATH.

  If you want to use a UNIX shell, then use MSYS Makefiles.
```

Any idea what I should do about that?

As for a License file: no, the original source doesn't contain a file - I'll ask the person mirroring the source on bitbucket (which is the source this recipe uses) if they can add one.

**Comment by ocefpaf** [comment]
AppVeyor is passing and that is used to create the Windows package here. We do need to fix Azure builds in the near future though b/c we are slowly migrating to Azure.

**Comment by robintw** [comment]
Ah excellent - I'll merge this then. Thanks.

**Pull request merged**